packages/astro/src/core/app/index.ts

```diff
@@ -131,9 +135,61 @@ export class App {
 		const url = new URL(request.url);
 		// ignore requests matching public assets
 		if (this.#manifest.assets.has(url.pathname)) return undefined;
-		const pathname = prependForwardSlash(this.removeBase(url.pathname));
-		const routeData = matchRoute(pathname, this.#manifestData);
-		// missing routes fall-through, prerendered are handled by static layer
+		let pathname;
+		if (this.#manifest.i18n && this.#manifest.i18n.routingStrategy === 'domain') {
+			// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host
+			let host = request.headers.get('X-Forwarded-Host');
+			// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto
+			let protocol = request.headers.get('X-Forwarded-Proto');
```

Comment:
This is another one coming from the proxy. So if it's not a proxy (or the proxy doesn't forward this error), we should have a fallback. We can use the protocol of the URL here.